test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,394 @@
+        # The next returned 3 non-zero values decreasing in magnitude, should match "expected" above.        
+        self.assertAlmostEqualVectorFixedPrecision(calculatedPitchSalience, expectedPitchSalience, 8)
+
+    def testSinglePeakLowCompression(self):
+        # Provide a single input peak with a 0.5 magnitude at the reference frequency
+        freq_speaks = [55]
+        mag_speaks = [0.5]
+        outputLength  = 600
+
+        # Low Compression expected values (0.0001)
+        expectedLowCompPitchSalience = [9.9993068e-01, 9.7546059e-01, 9.0444577e-01, 7.9383761e-01, 6.5446311e-01,
+        4.9996534e-01, 3.4546751e-01, 2.0609310e-01, 9.5484875e-02, 2.4470057e-02, 3.7491393e-33]
+
+        # Default magnitudeCompression expected values
+        expectedNormalPitchSalience = [5.0000000e-01, 4.8776412e-01, 4.5225424e-01, 3.9694631e-01, 3.2725424e-01,
+        2.5000000e-01, 1.7274573e-01, 1.0305370e-01, 4.7745746e-02, 1.2235877e-02, 1.8746997e-33]
+        
+        # Append zeros to expected saliences
+        expectedLowCompPitchSalience += [0] * (outputLength-11)
+        expectedNormalPitchSalience += [0] * (outputLength-11)
+
+        calculatedLowCompPitchSalience = PitchSalienceFunction(magnitudeCompression=0.0001)(freq_speaks, mag_speaks)
+        self.assertEqual(len(calculatedLowCompPitchSalience), outputLength)
+        self.assertAlmostEqualVectorFixedPrecision(calculatedLowCompPitchSalience, expectedLowCompPitchSalience, 8)
+
+        calculatedNormalPitchSalience = PitchSalienceFunction()(freq_speaks, mag_speaks)
+        self.assertEqual(len(calculatedNormalPitchSalience), outputLength)
+        self.assertAlmostEqualVectorFixedPrecision(calculatedNormalPitchSalience, expectedNormalPitchSalience, 8)
+
+    def testSinglePeakLowestMagThreshold(self):
+        # Provide a single input peak with a unit magnitude at the reference frequency,
+        freq_speaks = [55]
+        mag_speaks = [1]
+        outputLength  = 600
+
+        calculatedPitchSalience = PitchSalienceFunction(magnitudeThreshold=0)(freq_speaks, mag_speaks)
+        # With Mag. Threshold  = 0 outputs are zero
+        self.assertEqual(len(calculatedPitchSalience), outputLength)       
+        self.assertEqualVector(calculatedPitchSalience, zeros(600))        
+        """ FIXME
+        The intention here is to test if some peaks are correctly discarded, however, this test does not achieve this goal because there is only one input spectral peak.    
+        The magnitudeThreshold parameter defines the maximum allowed difference from the highest peak in dBs, so the input list of peaks should be of different magnitudes to see it working.
+        For example, to make it simple, we can add a few peaks with lower magnitudes than 1 and use magnitudeThreshold = 0. Then only the peak with the magnitude 1 will be selected and the current expectedPitchSalience will be still a correct expected test output
+        """
```

Comment:
This has to be addressed in the test. The current test does not make much utility.